include/cbl/CBLReplicator.h

```diff
@@ -148,6 +148,39 @@ typedef struct {
     FLString password;               ///< Password for proxy auth
 } CBLProxySettings;
 
+#ifdef COUCHBASE_ENTERPRISE
+
+/** Callback that encrypts \ref CBLEncryptable properties in documents pushed by the replicator.
+    @Note   If a null \ref FLSliceResult or an error is returned, the document will be failed to
+            replicate with the \ref kCBLErrorCrypto error when. For security reason, the encryption
+            cannot be skipped. */
+typedef FLSliceResult (*CBLPropertyEncryptor) (
+    void* context,              ///< Replicator’s context
+    FLString documentID,        ///< Document ID
+    FLDict properties,          ///< Document properties
+    FLString keyPath,           ///< Key path of the property to be encrypted
+    FLSlice input,              ///< Property data to be encrypted
+    FLString* algorithm,        ///< On return: algorithm name (Optional: Default Value is 'CB_MOBILE_CUSTOM')
```

Comment:
Shouldn't this be `FLSliceResult` since the caller is going to allocate it and pass ownership?

Reply:
Good point. This will not work if the string is dynamically created inside the callback code.

Reply:
Let's fix this in a separate PR (CBL-2299)